pkg/ccl/importccl/import_stmt.go

```diff
@@ -2058,18 +2069,26 @@ func (r *importResumer) Resume(ctx context.Context, execCtx interface{}) error {
 		return err
 	}
 
-	pkIDs := make(map[uint64]struct{}, len(details.Tables))
-	for _, t := range details.Tables {
-		pkIDs[roachpb.BulkOpSummaryID(uint64(t.Desc.ID), uint64(t.Desc.PrimaryIndex.ID))] = struct{}{}
+	pkIDs := make(map[uint64]int, len(details.Tables))
+	for i, t := range details.Tables {
+		pkIDs[roachpb.BulkOpSummaryID(uint64(t.Desc.ID), uint64(t.Desc.PrimaryIndex.ID))] = i
 	}
 	r.res.DataSize = res.DataSize
 	for id, count := range res.EntryCounts {
-		if _, ok := pkIDs[id]; ok {
+		if i, ok := pkIDs[id]; ok {
 			r.res.Rows += count
```

Comment:
It could definitely mislead the optimizer, although odds are it's probably still better than our current default of 1000 rows. Is there any way to tell whether the job has been paused? If so, it seems like we could use the following logic:
```
if job was never paused {
  // use row count from res 
} else {
  // use row count from res if > 1000
}
```
Obviously it would be ideal to persist the partial totals, but maybe this could be a a good stopgap in the mean time?

Reply:
I might take a stab at persisting partial results next week and then just make `res` correct on resumes too.

In the meantime, if you wanted to do that, I think you could do something like this, before calling ingest:
```
	var wasResumed bool
	initialProgress := r.job.Progress()
	for _, pos := range initialProgress.GetImport().ResumePos {
		wasResumed = wasResumed || pos > 0
	}
```
In any case, I don't think we need to persist the stats stubs in the job details -- we can just derive it at the end. Also, if/when we do persist something that changes during the execution of the job, we'd want to do so in progress, not details, to utilize the separate column family.

Reply:
Thank you, this is really helpful. I suspected this was the case, but wasn't totally sure. I will take a look at persisting counts in the progress field.

Reply:
@michae2 I _think_ there might be an existing bug here already -- caused by the addition of a retry loop around ingest -- that points to the fact we need to persist `res` _anyway_. If you want to just wait for us to do that, then rebase, I think you can use it once it is fixed without needing to persist anything related to stats in the job progress.

Reply:
That sounds great. I can rebase.

@rytaft do you think distinct and null counts are necessary? I was planning to work on those, too, but if they're not needed then I wouldn't have to touch the job details at all.

Reply:
I do think we'll want distinct counts and null counts eventually if possible. This would require storing the [hyperloglog sketch](https://github.com/cockroachdb/cockroach/blob/9afb59c302bc79bce6f469cfd72a6b7059bcdf82/pkg/sql/rowexec/sampler.go#L40) along with the running null count, but that shouldn't require too much space. We can use the `Sketch.MarshalBinary()` and `Sketch.UnmarshalBinary()` functions. Definitely no need for histograms.

For the first cut I think row count is enough, but adding distinct and null count will be important to further improve our estimates in the optimizer.

Reply:
Ah, right, sketches! So then `TableStatisticProto` is not even the correct message to persist. Ok, I will pull that out of job details for this PR, and wait for a future PR to add sketches and null counts to job progress. This has been so helpful! Thank you!

Reply:
Sounds good!

Reply:
> @michae2 I _think_ there might be an existing bug here already -- caused by the addition of a retry loop around ingest -- that points to the fact we need to persist `res` _anyway_. If you want to just wait for us to do that, then rebase, I think you can use it once it is fixed without needing to persist anything related to stats in the job progress.

@dt Just wondering, has anyone had a chance to work on this? (Looks like not yet, based on revision history?)

Reply:
Sorry, I don't think so -- I've been OOO and I don't think anyone else picked it up yet.

Reply:
That's fine, it's not urgent.

Reply:
@michae2, just wanted to mention that I've filed https://github.com/cockroachdb/cockroach/issues/67987 to track this.